### ISSUE TITLE
Malformed attributes cause ClassCastException in XML Parser

### ISSUE THREAD

**Issue by comtef** [databricks/spark-xml#260]
XML documents with malformed attributes cannot be parsed. 
It cause a ClassCastException and the Spark job will fail.

Example document :
```xml
<book id="Malformed attribute with " caracter ">
      <author>Kress, Peter</author>
      <title>Paradox Lost</title>
      <genre>Science Fiction</genre>
      <price>6.95</price>
      <publish_date>2000-11-02</publish_date>
      <description>After an inadvertant trip through a Heisenberg
      Uncertainty Device, James Salway discovers the problems
      of being quantum.</description>
</book>
```

Stacktrace :

```
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 144.0 failed 1 times, most recent failure: Lost task 0.0 in stage 144.0 (TID 151, localhost): java.lang.ClassCastException: com.sun.xml.internal.stream.events.StartDocumentEvent cannot be cast to javax.xml.stream.events.StartElement
[info]  at com.sun.xml.internal.stream.events.DummyEvent.asStartElement(DummyEvent.java:129)
[info]  at com.databricks.spark.xml.util.InferSchema$$anonfun$3$$anonfun$apply$2.apply(InferSchema.scala:99)
[info]  at com.databricks.spark.xml.util.InferSchema$$anonfun$3$$anonfun$apply$2.apply(InferSchema.scala:89)
[info]  at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
[info]  at scala.collection.Iterator$class.foreach(Iterator.scala:742)
[info]  at scala.collection.AbstractIterator.foreach(Iterator.scala:1194)
[info]  at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:155)
[info]  at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1194)
[info]  at scala.collection.TraversableOnce$class.aggregate(TraversableOnce.scala:212)
[info]  at scala.collection.AbstractIterator.aggregate(Iterator.scala:1194)
[info]  at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1$$anonfun$24.apply(RDD.scala:1093)
[info]  at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1$$anonfun$24.apply(RDD.scala:1093)
[info]  at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1$$anonfun$25.apply(RDD.scala:1094)
[info]  at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1$$anonfun$25.apply(RDD.scala:1094)
[info]  at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:766)
[info]  at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:766)
[info]  at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
[info]  at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:319)
[info]  at org.apache.spark.rdd.RDD.iterator(RDD.scala:283)
[info]  at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:70)
[info]  at org.apache.spark.scheduler.Task.run(Task.scala:85)
[info]  at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)
[info]  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]  at java.lang.Thread.run(Thread.java:748)
```

**Comment by HyukjinKwon** [comment]
Sounds we should be able to permissively read by the parse mode.

**Comment by comtef** [comment]
I've posted a PR to fix this

**Comment by HyukjinKwon** [comment]
This was resolved in #261.

**Comment by dalwindr** [comment]
i am getting same error my code is in scala... not working at all
